connector/src/main/java/org/geysermc/connector/network/translators/collision/CollisionManager.java

```diff
@@ -262,6 +260,7 @@ public boolean isUnderSlab() {
             playerBoundingBox.setSizeY(EntityType.PLAYER.getHeight());
             playerBoundingBox.setMiddleY(standingY);
             boolean result = collision.checkIntersection(playerBoundingBox);
+            collision.setPosition(null);
```

Comment:
Yeah, I see what you mean about concurrency issues, but it's something that needs to be addressed anyway. I'll add it to my TODO list in the future so let me know if you tackle it instead.